packages/shared/src/db/queries.ts

```diff
@@ -1205,6 +1208,39 @@ export const setJoinedGroupChannels = createWriteQuery(
   ['channels']
 );
 
+export const addJoinedGroupChannel = createWriteQuery(
+  'addJoinedGroupChannel',
+  async ({ channelId }: { channelId: string }, ctx: QueryCtx) => {
+    logger.log('addJoinedGroupChannel', channelId);
+
+    await ctx.db.insert($groupNavSectionChannels).values({
+      channelId,
+      groupNavSectionId: 'default',
+    });
+
+    return await ctx.db
+      .update($channels)
+      .set({
+        currentUserIsMember: true,
+      })
+      .where(eq($channels.id, channelId));
+  },
+  ['channels']
+);
+
+export const removeJoinedGroupChannel = createWriteQuery(
+  'removeJoinedGroupChannel',
+  async ({ channelId }: { channelId: string }, ctx: QueryCtx) => {
```

Comment:
Could this be combined with `setLeftGroupChannels`?

Reply:
Same comments apply as in the setJoinedGroupChannels case.